sdk/core/core-amqp/src/errors.ts

```diff
@@ -464,9 +468,23 @@ export class MessagingError extends Error {
   info?: any;
   /**
    * @param {string} message The error message that provides more information about the error.
+   * @param originalError An error whose properties will be copied to the MessagingError if the
+   * propert doesn't already exist.
```

Comment:
Why copy only if the property already exists and not copy all? They will get overriden appropriately after the constructor call right?

Reply:
Since the properties are copied after the `super` call, we don't want to end up overwriting the `message`, `name`, and `stack` properties with those from the original error. This also means if the error had `translated` or `retryable` properties already, they would not overwrite those values set in the constructor (`true` for both).

Maybe we should only update the `retryable` field if the original error doesn't have one (if someone passes an error with this already set should we honor it?). Since `message` is passed in explicitly, `name` is always `MessagingError`, and `stack` should indicate the line where this error is created, I don't think those should be overwritten.

Reply:
Also, should be noted that this copies properties if they __don't__ exist on `MessagingError`, not if they do.

Reply:
I made a late edit that it might be better to copy the stack over as well, so I'll go ahead and make that change.